### PR TITLE
NEW: Added CursorLockBehavior to InputSystemUIInputModule to enable pointer events when the cursor is locked.

### PR DESCRIPTION
**Pull request by andrew-oc**
[Fogbugz](https://fogbugz.unity3d.com/f/cases/1395281/)
[Issue Tracker](https://issuetracker.unity3d.com/product/unity/issues/guid/1395281/)

### Description

IPointerEnterHandler and IPointerExitHandler interfaces did not work when the cursor was locked. This made it impossible to use PhysicsRaycaster to interact with world space UI or objects using pointer-driven gaze direction. The issue occurs because when the cursor is locked, the raycast origin is placed outside the screen, which causes raycast logic to short-circuit and always miss all objects.

### Changes made

Added the CursorLockBehaviour enum and a setting in InputSystemUIInputModule to allow the user to specify that when the cursor is locked, raycasts should originate from the center of the screen.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs [hunk 1]
```diff
@@ -167,6 +168,14 @@ public override void OnInspectorGUI()
                 EditorGUI.EndDisabledGroup();
             }
 
+            m_AdvancedFoldoutState = EditorGUILayout.Foldout(m_AdvancedFoldoutState, new GUIContent("Advanced"), true);
+            if (m_AdvancedFoldoutState)
+                EditorGUILayout.PropertyField(serializedObject.FindProperty("m_CursorLockBehavior"),
+                    EditorGUIUtility.TrTextContent("Cursor Lock Behavior",
+                        $"Controls the origin point of UI raycasts when the cursor is locked. {InputSystemUIInputModule.CursorLockBehavior.OutsideScreen} " +
+                        $"is the default behavior and will force the raycast to miss all objects. {InputSystemUIInputModule.CursorLockBehavior.ScreenCenter} " +
+                        $"will cast the ray from the center of the screen."));
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs:177]: Should probably update manual screenshot and doc.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs [hunk 2]
```diff
@@ -82,6 +82,24 @@ public UIPointerBehavior pointerBehavior
             set => m_PointerBehavior = value;
         }
 
+        /// <summary>
+        /// Where to position the pointer when the cursor is locked.
+        /// </summary>
+        /// <remarks>
+        /// By default, the pointer is positioned at -1, -1 in screen space when the cursor is locked. This has implications
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs:89]: I know I asked this before but just to make 100% extra super sure... `StandaloneInputModule` does not do the raycast-from-center thing? If it does, default is probably better the other way round.

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModuleEditor.cs:177]: Agree, and include @duckets on the review after the change

**Review by ekcoh — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs (hunk 2 above)
Comment (andrew-oc) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs:89]: Double checked. StandaloneInputModule also sets the raycast origin to -1, -1.

Comment (Rene-Damm) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs:89]: Cool, thank you :)

**Review by Rene-Damm — APPROVED** [review]

**Comment by K-A-C** [comment]
I'm still having issues with getting a locked cursor to interact with worldspace UI even with this fix implemented. Unlocking the cursor works fine so I know that my UI is okay. I've set the `CursorLockBehavior` to `Screen Center` on the Input system UI Input Module

**Pull request merged**